include/eve/product_type.hpp

```diff
@@ -156,10 +157,19 @@ namespace eve
   {
     using tuple_type = kumi::tuple<Fields...>;
 
-    EVE_FORCEINLINE friend auto operator+(eve::like<Self> auto x, eve::like<Self> auto y) requires requires { x += y; }
+    template<like<Self> Z1, like<Self> Z2>
+    EVE_FORCEINLINE friend auto operator+(Z1 x, Z2 y) requires requires { x += x; }
     {
-      x += y;
-      return x;
+      if constexpr(scalar_value<Z1> && simd_value<Z2>)
+      {
+        as_wide_t<Z1> that{x};
+        return that += y;
```

Comment:
Why not just y += x;

Reply:
Good point

Reply:
Note that it will be needed for - and / that are no symmetric